src/app/detail/ui/TaskEditor.tsx

```diff
@@ -163,7 +165,12 @@ export const TaskEditor = ({
           readonly={userType === UserType.CLIENT_USER}
           editorClass="tapwrite-details-page"
           placeholder="Add description..."
-          uploadFn={(file) => uploadImageHandler(file, token ?? '', task_id)}
+          uploadFn={async (file) => {
+            setIsUserTyping(true)
+            const url = await uploadImageHandler(file, token ?? '', task_id)
+            setIsUserTyping(false)
```

Comment:
What if the user types a sentence while upload is in progress?

Looking at the previous implementation of `handleDetailChange` I noticed we could have edge cases here so I tested it out... image upload disappears if I type something while waiting for the image to upload.

[Screencast from 2024-10-09 18-19-20.webm](https://github.com/user-attachments/assets/7cd9e4bd-4e81-4a59-a1c3-ed0fb93d2bf7)

Reply:
As i mentioned in the slack message, there are other consequences to this as well. Another one I can think of is uploading two or more images, where settings isUserTyping to false from one callback will affect the rest.